Created Laser Rifle, Fixed and Updated Laser Pistol and Character System

Alasses opened the pull request:
1. Set the title to exactly what you are adding.

2. Which team is this for?
Mechanics - Character System

3. Give a short description of what you are implementing:
- [Update] Update the weapon system to fit the networking requirement, directly host the game without building the game won't work due to the port issues, please build the game if you want to run the game host.
- [Update] Update the old laser gun into the laser pistol, to fit the networking requirement, this is not  a new weapon
- [Feature] Create a new weapon called laser rifle, along with a new UI for the weapon.
- [Bug] Fixed the gravity issues for suddenly hit the ground when walking off the roof. This does not conflict with current work assigned to other people since the gravity equation wasn't removed, only one new line check was added into the script, please read the script carefully.
- [Bug] Fixed the character rotation issue to rotate the whole character. The character was now separate into head and body which act differently when the player tries to rotate the camera, only the head and weapon will totally rotate now, the body will keep standing on the group now.
- [Bug] Fixed the bug that the ranged weapon won't trigger if shooting at nothing once.

Alasses committed:
Laser Rifle V1.0

Squashed commit:

[703d1e6] Fixed the null reference of the UI

[58afe59] Update the scene

[c6b3d56] Fixed the body rotation

- Separate the player into head and body
- Separate the rotation into head rotation and body rotation

[c9343b2] Remove the test build

- Rolled back the git ignore file
- Removed the test folder

[49f4d12] Fixed bug for range weapon

- Fixed the bug that when shooting the weapon towards nothing it won't trigger anymore.

[ae14d13] Update .gitignore

[89e893f] Fixed the laser pistol

- Update the laser pistol to make it work for current system.

[609a6cb] Laser Rifle v1.0

- Fixed network issues of the weapon system
- Created a basic networking weapon structure

[ff848a3] Laser Rifle v0.4

- Successfully reverted to old commit and fixed problem
- Fixed jump bug

[5cb3a90] Revert to v0.3

[73114d2] Revert to v0.2

[25ef25f] Revert to v0.3

[029a6e5] Revert to v0.3

Everything fucked up, trying to fix them

[bb1795b] Revert to v0.2

Everything fucked up, trying to fix them

[5b2e398] Fail build

[f1ddfe3] Laser Rifle v0.3

- Modified the weapon and ammo system to fit the networking requirements

[96289c4] Laser Rifle v0.2

- Fixed the UI loading issues (+1 squashed commits)

Squashed commits:

[aeb2d80] Laser Rifle v0.1

- Created basic Laser Rifle
- Modified the structure of weapon system
- Modified the ammo classes

Alasses committed:
Laser Rifle v0.1

- Created basic Laser Rifle
- Modified the structure of weapon system
- Modified the ammo classes

Laser Rifle v0.2

- Fixed the UI loading issues

Laser Rifle v0.3

- Modified the weapon and ammo system to fit the networking requirements

Fail build

Revert to v0.2

Everything fucked up, trying to fix them

Revert to v0.3

Everything fucked up, trying to fix them

Revert to v0.3

Revert to v0.2

Revert to v0.3

Laser Rifle v0.4

- Successfully reverted to old commit and fixed problem
- Fixed jump bug

Laser Rifle v1.0

- Fixed network issues of the weapon system
- Created a basic networking weapon structure

Fixed the laser pistol

- Update the laser pistol to make it work for current system.

Update .gitignore

Remove the test build

- Rolled back the git ignore file
- Removed the test folder

Fixed the body rotation

- Separate the player into head and body
- Separate the rotation into head rotation and body rotation

Update the scene

Fixed the null reference of the UI

Removed player instance from the scene

Fixed bug for range weapon

- Fixed the bug that when shooting the weapon towards nothing it won't trigger anymore.

Alasses committed:
Laser Rifle v0.1

- Created basic Laser Rifle
- Modified the structure of weapon system
- Modified the ammo classes

Alasses committed:
Laser Rifle v0.2

- Fixed the UI loading issues

Alasses committed:
Laser Rifle v0.3

- Modified the weapon and ammo system to fit the networking requirements

Alasses committed:
Fail build

Alasses committed:
Revert to v0.2

Everything fucked up, trying to fix them

Alasses committed:
Revert to v0.3

Everything fucked up, trying to fix them

Alasses committed:
Revert to v0.3

Alasses committed:
Revert to v0.2

Alasses committed:
Revert to v0.3

Alasses committed:
Laser Rifle v0.4

- Successfully reverted to old commit and fixed problem
- Fixed jump bug

Alasses committed:
Laser Rifle v1.0

- Fixed network issues of the weapon system
- Created a basic networking weapon structure

Alasses committed:
Fixed the laser pistol

- Update the laser pistol to make it work for current system.

Alasses committed:
Update .gitignore

Alasses committed:
Remove the test build

- Rolled back the git ignore file
- Removed the test folder

Alasses committed:
Fixed the body rotation

- Separate the player into head and body
- Separate the rotation into head rotation and body rotation

Alasses committed:
Update the scene

Alasses committed:
Fixed the null reference of the UI

Alasses committed:
Removed player instance from the scene

Nathan127 commented:
Could you set this to merge to dev instead of master?

Alasses commented:
Oh sorry my fault

JasonStallkamp commented:
I think before we merge this we should do a squash. This will help clean up the commits and the files. I can walk you through it at the next meeting.

Alasses commented:
> I think before we merge this we should do a squash. This will help clean up the commits and the files. I can walk you through it at the next meeting.

Of course, especially for the new networking stuff.

Alasses committed:
Fixed bug for range weapon

- Fixed the bug that when shooting the weapon towards nothing it won't trigger anymore.

Alasses committed:
Merge branch 'Feature-Mechanics-LaserRifle' of https://github.com/OSUGameDev/LocalWar into Feature-Mechanics-LaserRifle

JasonStallkamp commented:
Local War/UIElementsSchema/UnityEditor.PackageManager.UI.xsd
do you know what this file is and why it was included

JasonStallkamp commented:
Local War/Assets/Scene/Arena.unity
Why did you touch this file. I wouldnt think your changes would touch the map.

JasonStallkamp commented on Local War/Assets/Resources/Script/Weapon/Weapon.cs:
```diff
@@ -1,20 +1,55 @@
+    protected int           type;
+    protected string        description;
+    protected WeaponSys     playerWeaponSys;
+    protected bool          isShooting;
+    protected bool          isFinishShoot;
```
Why is shoot in weapon. Weapon should not have a concept of shooting. It could be either a ranged weapon or a melee weapon. All of this code could and probably should go into ranged weapon.

Nathan127 committed:
Merge branch 'Feature-Mechanics-LaserRifle' of https://github.com/OSUGameDev/LocalWar into Feature-Mechanics-LaserRifle

moorbren commented:
Closing this. Tried merging this with the main project, but it is an ENORMOUS pain. It would be significantly less work now to just rewrite the code.
